src/mlang/backend_ir/bir.ml

```diff
@@ -52,7 +69,10 @@ let rec get_block_statements (rules : rule RuleMap.t) (stmts : stmt list) :
 
 (** Returns program statements with all rules inlined *)
 let get_all_statements (p : program) : stmt list =
-  get_block_statements p.rules p.statements
+  FunctionMap.fold
+    (fun _ mpp_function stmts ->
+      get_block_statements p.rules mpp_function @ stmts)
+    p.mpp_functions []
```

Comment:
This wasn't put in the comment but this function is supposed to return the program completely inlined, as in before functions and rules was a thing. Here the function inlines the rules at call sites but concatenates the functions bodies disregarding the program semantics.
I'm not sure where this function is still used but it will probably break things somewhere.